app/main.py

```diff
@@ -1,8 +1,40 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(
+            self,
+            name: str,
+            age: int
+    ) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    for person in people:
+        Person(person["name"], person["age"])
+
+    print(Person.people)
+
+    for person in people:
+        instance = Person.people[person["name"]]
+
+        if "wife" in person and person["wife"]:
+            instance.wife = Person.people[person["wife"]]
+        elif "husband" in person and person["husband"]:
+            instance.husband = Person.people[person["husband"]]
+
+    result = []
+
+    for person in people:
+        result.append(Person.people[person["name"]])
+
```

Comment:
it's better to create a list of persons in the first loop, so you don't need to loop through all persons twice